File: src/Telegram.ServerApp/Telegram.Domain/Entities/ChatRoom.cs
using Telegram.Domain.Common.Entities;

namespace Telegram.Domain.Entities;

public class ChatRoom : AuditableEntity
{
    public long FirstUserId { get; set; }
    public User FirstUser { get; set; }

    public long SecondUserId { get; set; }
    public User SecondUser { get; set; }

    public int FirstUserUnreadMessageCount { get; set; }
    public int SecondUserUnreadMessageCount { get; set; }

    public long? LastMessageId { get; set; }
    public Message? LastMessage { get; set; }

    public IEnumerable<Message> Messages { get; set; }
}

File: src/Telegram.ClientApp/src/modules/models/ChatRoom.ts
import type { Message } from "./Message";
import type { User } from "./User";

export interface ChatRoom {
    id: number;

    firstUserId: number;
    firstUser: User;

    secondUserId: number;
    secondUser: User;

    lastMessageId: number;
    lastMessage: Message | null;

    firstUserUnreadMessageCount: number;
    secondUserUnreadMessageCount: number;
}